gobblin-utility/src/main/java/gobblin/util/callbacks/CallbacksDispatcher.java

```diff
@@ -43,6 +45,7 @@
 public class CallbacksDispatcher<L> {
   private final Logger _log;
   private final List<L> _listeners = new ArrayList<>();
+  private final List<WeakReference<L>> _autoListeners = new ArrayList<>();
```

Comment:
Why a hash map? What are the values of the map? I think managing the weakreferences is trivial.